app/src/main/java/ch/epfl/sdp/appart/AdActivity.java

```diff
@@ -51,6 +53,7 @@ protected void onCreate(Bundle savedInstanceState) {
         mViewModel.getPrice().observe(this, this::updatePrice);
         mViewModel.getDescription().observe(this, this::updateDescription);
         mViewModel.getAdvertiser().observe(this, this::updateAdvertiser);
+        mViewModel.getAdvertiserId().observe(this, this::updateAdvertiserId);
```

Comment:
Why is this needed?

Reply:
In order to set the advertiserId as soon as the ad is initialised. Another approach could be to simply store a String without observing it but this would imply calling the view model from the openContactInfo once the add is already initialised. I thought it would be cleaner to set this attribute as the others, but we could always change it if there is a better way